compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsAnydataUniqueVisitor.java

```diff
@@ -70,7 +70,10 @@ private boolean isAnydata(BType type) {
             case TypeTags.BOOLEAN:
             case TypeTags.JSON:
             case TypeTags.XML:
-            case TypeTags.TABLE:
+            case TypeTags.XML_TEXT:
+            case TypeTags.XML_ELEMENT:
```

Comment:
Is it correct to add `JSON` here? (In L71)(not related to this issue)
By that, we allow the following kind of scenario also.
```ballerina
public function main() {
        table<map<json>> tab3 = table [ ];
    
        anydata l = tab3;
}
```

Reply:
Why should this be disallowed?

Reply:
is `table<map<json>>` is sub type of `anydata`. ?
Spec says `anydata can be defined as () | boolean | int | float | decimal | string | XML | anydata[] | map<anydata> | table<map<anydata>>` (Is `json` implies from these types)

Reply:
`json` is `() | boolean | int | float | decimal | string | json[] | map<json>`.

`anydata` is `() | boolean | int | float | decimal | string | xml | anydata[] | map<anydata> | table<map<anydata>>`.

The set of shapes represented by `json` is a subtype of that represented by `anydata`. Therefore `json` is a subtype of `anydata`.